### DISCUSSION TITLE
Device 'awa_spi' is disabled due to an error

### DISCUSSION THREAD

**Discussion by easaw** [awawa-dev/HyperSPI#52] [Q&A] [answered]
I get the error:

`Failed to open device (/dev/spidev0.0). Error message: No such file or directory`

I have tried ESP32 (D1 R32), ESP32 Seeduino, and  ESP32-S2 Mini (2 devices). It never connects, it shows in the logs, but just disables it.

'spidev' is not in `/dev` - do I need to install something?

I've tried this many times and it's getting really frustrating, I'm not sure what to do :C

I'm on a Pi 4 with Pi OS Bookworm 64bit Lite (not that it should make much difference).

WLED runs fine, and doesn't even need a level shifter on the D1.

I'm just stuck with this. The [HyperSerialESP32](https://github.com/awawa-dev/HyperSerialESP32) is also causing me issues where I can't get the USB device. It never sees it. I can't see what it would be in `/dev`

_I haven't tried HyperSerialWLED, because I don't want to use it._

**Comment by awawa-dev** [comment, marked as answer]
Hmm...but you know that if you dont use our SD OS image or HyperHDR installer (that also can do the trick but the restart is required) then spidev0.0 is disabled on default? Use raspi-config and reboot.

**Comment by easaw** [comment]
Sorry, what do I use "Use raspi-config" to enable? Thank you.

I used the method here: https://github.com/awawa-dev/HyperHDR/wiki/Installation#debian-and-ubuntu-apt-repository to install.

If this doesn't work I'll try the SD card, but I prefer to install the entire OS directly from RPi (no offence!)

**Reply by awawa-dev** [reply]
and did you restart RPi after that?

**Reply by easaw** [reply]
Don't worry, I found the 'enable SPI' option 🤦‍♂️

Haven't tested but I assume that was always the problem. I've been going nuts.
Can't even blame this level of ignorance on the ADHD lol...
